File: 1er_parcial_recuperatorio/python/modelos/Bicicleta.py
from modelos.Transporte import Transporte


class Bicicleta(Transporte):

    def __init__(self, nombre) -> None:
        self.__nombre = nombre
        self.__peso_max = 5  # KG
        self.__velocidad_promedio = 12  # Km/h
        self.__valor_por_km = 200  # pesos

    def transportar_paquete(self, distancia, pesoPaquete):
        if (pesoPaquete > self.__peso_max):
            return -1

        indicador_conveniencia = 0.2 * self.tiempo_estimado(distancia) + 0.8 * \
            self.costo_estimado(distancia)

        return indicador_conveniencia

    @property
    def nombre(self):
        return self.__nombre

    def tiempo_estimado(self, distancia):
        return distancia / self.__velocidad_promedio  # Horas

    def costo_estimado(self, distancia):
        return distancia * self.__valor_por_km

File: 1er_parcial_recuperatorio/java/PrimerParcialRecuperatorio/src/modelos/Bicicleta.java
package modelos;

public class Bicicleta implements Transporte {

    private String nombre;
    private final double pesoMax = 5; // KG
    private final double velocidadPromedio = 12; // Km/h
    private final double valorPorKm = 200; // pesos

    public Bicicleta(String nombre) {
        this.nombre = nombre;
    }

    public double transportarPaquete(double distancia, double pesoPaquete) {
        if (pesoPaquete > pesoMax) {
            return -1;
        }

        double indicadorConveniencia = 0.2 * tiempoEstimado(distancia) + 0.8 * costoEstimado(distancia);

        return indicadorConveniencia;
    }

    public String getNombre() {
        return nombre;
    }

    public double getPesoMax() {
        return pesoMax;
    }

    public double tiempoEstimado(double distancia) {
        return distancia / velocidadPromedio; // Horas
    }

    public double costoEstimado(double distancia) {
        return distancia * valorPorKm;
    }
}
